### ISSUE TITLE
Add smaller optimization for do while loops

### ISSUE THREAD

**Issue by thomasd16** [google/closure-compiler#788]
If you enter the following under advanced compilation level:
```javascript
do {
	something(i);
} while (i--);

//You get the following code:
do something(i);while(i--);

//However it would be nice if it would compile down to the following:
 while(something(i),i--);

//This does exactly the same thing while being considerably shorter.

// You can also do similar compression for regular while loops
// using similar syntax however the savings are much smaller:
while (i--) {
	something(i);
}

//could become
for(;i--;something());
```
There may be some reason I missed for why this is not implemented weather or not there are fringe cases in witch it changes what the code does.

**Comment by MatrixFrog** [comment]
My intuition (which could be totally wrong of course) is that do/while loops are used rarely enough that it's not worth it to worry about a few bytes on each one. The change for while loops might be cool though.

**Comment by thomasd16** [comment]
Actually I did my math wrong for the plain while loops doing `for(;condition;expression);` takes the same space as `for(;condition;)expression;`

`while(expression,conditiong);` vs `do expression;while(condition);` does save 3 bites. And could be used for any block that boils down to an expression. (No returns, throws, continues, breaks, or nested loops).

Also your intuition appears to be correct. I scanned 2 projects for do while loops jQuery and UglifyJS the former had none and the latter only had 3.
